code/modules/food&drinks/food/snacks.dm

```diff
@@ -151,7 +152,12 @@
 			return 1
 
 //Called when you finish tablecrafting a snack.
-/obj/item/weapon/reagent_containers/food/snacks/CheckParts()
+/obj/item/weapon/reagent_containers/food/snacks/CheckParts(list/content)
+	..()
+	reagents.reagent_list.Cut()
+	for(var/obj/item/weapon/reagent_containers/RC in contents)
+		RC.reagents.trans_to(reagents, RC.reagents.maximum_volume)
```

Comment:
You also need to make sure that items that were in reqs, but not parts originally, do not end up inside the result object.